File: 백준/Silver/1193. 분수찾기/분수찾기.cc
#include <iostream>
using namespace std;

int main() {
    int x;
    cin >> x;

    int line = 1;

    while (x > line) {
        x -= line;
        line++;
    }

    if (line % 2 == 0) {
        cout << x << "/" << line - x + 1;
    } else {
        cout << line - x + 1 << "/" << x;
    }

    return 0;
}

File: 백준/Silver/1193. 분수찾기/분수찾기.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int x = sc.nextInt();

        int line = 1;

        while (x > line) {
            x -= line;
            line++;
        }

        if (line % 2 == 0) {
            System.out.println(x + "/" + (line - x + 1));
        } else {
            System.out.println((line - x + 1) + "/" + x);
        }
    }
}